### PR TITLE
vimc-4306: Ensure spec is up to date

### PR DESCRIPTION
**Pull request by r-ash**
* Updates the spec including adding notes about workflow endpoints and new statuses
* Add a PR template to remind us to keep it up to date in the future

### CODE REVIEW

File: inst/schema/spec.md [hunk 1]
```diff
@@ -67,12 +80,14 @@ Schema: [`Status.schema.json`](Status.schema.json)
         {
             "key": "antiutopian_peregrinefalcon",
             "status": "running",
-            "name": "minimal"
+            "name": "minimal",
+            "version": 20210211-143212-98c45632
```
Comment (richfitz) [inline inst/schema/spec.md:84]: This is invalid json and requires quotes (see below too)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- [ ] spec.md has been updated or doesn't need to updated
```
Comment (richfitz) [inline .github/pull_request_template.md:1]: Is there anything we can add that will process the spec programmatically? I feel we've done this elsewhere. Alternatively  we should add support in porcelain for doing openapi/swagger as that will allow examples to be included and tested more easily

File: inst/schema/spec.md [hunk 3]
```diff
@@ -395,3 +412,195 @@ Response schema: [`ReportInfo.schema.json`](ReportInfo.schema.json)
...
+          "[ git        ]  fetch",
+          "[ checkout   ]  579dd44fa71f579cbe5669578718c957015a71f3; was master",
+          "[ name       ]  minimal",
+          "[ id         ]  20220211-171617-6b2b2cac",
+          "[ id_file    ]  /tmp/RtmpUyRoEL/file30698f4fe64e83/runner/id/chartreuse_alleycat.id_file",
+          "[ start      ]  2022-02-11 17:16:17",
+          "[ git        ]  checkout master; was HEAD",
+          "[ data       ]  source => dat: 20 x 2",
+          "",
+          "> png("mygraph.png")",
```
Comment (richfitz) [inline inst/schema/spec.md:554]: Escape quotes

**Review by richfitz — CHANGES_REQUESTED** [review]
There's a bigger question here about how we keep this accurate - I don't expect that these responses all are current or correct and there's no real way of knowing. It would be good to automate testing this, but that would be probably best done in porcelain?  Happy to merge this once the small changes are made but would be curious for your thoughts on a longer term solution

**Comment by r-ash** [comment]
Agree would be good if we had some testing that this was correct, I also expect it is not up to date and as we've seen is hard to remember to keep it up to date! Having some swagger documentation with examples would be a good step there. I also like the look of [pact](https://docs.pact.io/) which is based around contract testing but generates docs for you too. It would be more like orderly-web describes the contract that orderly.server needs to provide and then we have contract tests which make sure our API meets their needs. I think focused on teams who might have multiple consumers of their API instead of us where we just have 1 so might be overkill. But would probably simplify testing for web devs.

**Review by richfitz — APPROVED** [review]

**Pull request merged**